Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,423 @@
+		                                  const uint8_t mac2[16]);
+
+		uint32_t getMessageType() const;
+		const uint8_t* getReserved() const;
+		uint32_t getSenderIndex() const;
+		const uint8_t* getInitiatorEphemeral() const;
+		const uint8_t* getEncryptedInitiatorStatic() const;
+		const uint8_t* getEncryptedTimestamp() const;
+		const uint8_t* getMac1() const;
+		const uint8_t* getMac2() const;
+
+		wg_handshake_initiation* getHandshakeInitiationHeader() const
+		{
+			return (wg_handshake_initiation*)getBasicHeader();
+		}
+
+		// implement abstract methods
+
+		WireGuardMessageType getWireGuardMessageType() const
+		{
+			return WireGuardMessageType::HandshakeInitiation;
+		}
+	};
+
+	/**
+	 * @class WireGuardHandshakeResponseLayer
+	 * Represents a Handshake Response message
+	 */
+	class WireGuardHandshakeResponseLayer : public WireGuardLayer
+	{
+	public:
+#pragma pack(push, 1)
+		/**
+		 * @struct wg_handshake_response
+		 * Represents the Handshake Response message structure
+		 */
+		typedef struct wg_handshake_response : wg_common_header
+		{
+			/** Sender index */
+			uint32_t senderIndex;
+			/** Receiver index */
+			uint32_t receiverIndex;
+			/** Responder's ephemeral public key */
+			uint8_t responderEphemeral[32];
+			/** Encrypted empty field */
+			uint8_t encryptedEmpty[16];
+			/** MAC1 field */
+			uint8_t mac1[16];
+			/** MAC2 field */
+			uint8_t mac2[16];
+		} wg_handshake_response;
+#pragma pack(pop)
+
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		WireGuardHandshakeResponseLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+		    : WireGuardLayer(data, dataLen, prevLayer, packet)
+		{}
+
+		/**
+		 * A constructor that creates a new Handshake Response message
+		 * @param[in] senderIndex The sender index
+		 * @param[in] receiverIndex The receiver index
+		 * @param[in] responderEphemeral The responder's ephemeral public key
+		 * @param[in] encryptedEmpty The encrypted empty field
+		 * @param[in] mac1 The MAC1 field
+		 * @param[in] mac2 The MAC2 field
+		 */
+		WireGuardHandshakeResponseLayer(uint32_t senderIndex, uint32_t receiverIndex,
+		                                const uint8_t responderEphemeral[32], const uint8_t encryptedEmpty[16],
+		                                const uint8_t mac1[16], const uint8_t mac2[16]);
+
+		uint32_t getMessageType() const;
+		const uint8_t* getReserved() const;
+		uint32_t getSenderIndex() const;
+		uint32_t getReceiverIndex() const;
+		const uint8_t* getResponderEphemeral() const;
+		const uint8_t* getEncryptedEmpty() const;
+		const uint8_t* getMac1() const;
+		const uint8_t* getMac2() const;
```

Comment:
Maybe these properties should return `std::array` and copy the data from the packet to the array?

Same in properties in the other classes that return `uint8_t*`